base/uk.ac.stfc.isis.ibex.ui.journalviewer/src/uk/ac/stfc/isis/ibex/ui/journalviewer/models/JournalViewModel.java

```diff
@@ -44,202 +44,373 @@
-    }
-    
-    /**
-     * Gets whether a particular field is currently selected.
-     * @param field the field to get
-     * @return true if the field is selected, false otherwise
-     */
-    public boolean getFieldSelected(JournalField field) {
-    	return model.getSelectedFields().contains(field);
-    }
-    
-    /**
-     * @return the sort direction of the primary sort for SWT to use
-     */
-    public int getSortDirection() {
-        if (getActiveSearch().getPrimarySort().getDirection() == JournalSortDirection.DESCENDING) {
-            return SWT.DOWN;
-        } else {
-            return SWT.UP;
-        }
-    }
-    
-    private String dateToString(Date lastUpdate) {
-        if (lastUpdate == null) {
-            return "N/A";
-        }
-        if (isToday(lastUpdate)) {
-            return DateFormat.getTimeInstance(DateFormat.MEDIUM).format(lastUpdate);
-        } else {
-            return DateFormat.getDateTimeInstance().format(lastUpdate);
-        }
-    }
-
-    /**
-     * @param date Is the provided date the same as today's date
-     */
-    private boolean isToday(Date date) {
-        Calendar calendar = Calendar.getInstance();
-        calendar.set(Calendar.HOUR_OF_DAY, 0);
-        calendar.set(Calendar.MINUTE, 0);
-        calendar.set(Calendar.SECOND, 0);
-        Date today = calendar.getTime();
-        return date.after(today);
-    }
-    
-    /**
-     * @param pageNumber The number of the results page.
-     * @return a CompletableFuture
-     */
-    public CompletableFuture<Void> setPageNumber(int pageNumber) {
-        return model.setPage(pageNumber);
-    }
-    
-    /**
-     * @return The current journal entries page.
-     */
-    public int getPageNumber() {
-    	return model.getPage();
-    }
-    
-    /**
-     * @param max The maximum number of pages supported by the journal view.
-     */
-    public void setPageNumberMax(int max) {
-	    firePropertyChange("pageNumberMax", 0, model.getPageMax());
-    }
-    
-    /**
-     * @return The maximum number of pages supported by the journal view.
-     */
-    public int getPageNumberMax() {
-    	return model.getPageMax();
-    }
-    
-    /**
-     * @return the searchableFields
-     */
-    public List<JournalField> getSearchableFields() {
-        return model.getSearchableFields();
-    }
-
-    /**
-     * Sorts by the specified field, and swaps the direction if already active.
-     * @param field The field to sort by
-     * @return a CompletableFuture
-     */
-    public CompletableFuture<Void> sortBy(JournalField field) {
-        return model.sortBy(field);
-    }
-    
+	private JournalModel model;
+	private String message;
+	private List<JournalRow> runs;
+	private String lastUpdate;
+	private int toNumber = 0;
+	private int fromNumber = 0;
+	private String noError = "";
+	private String runNumberError = "Run number in incorrect order";
+	private String dateTimeError = "Date/Time in wrong order";
```

Comment:
If these are constants, they should be declared `private static final String ...`, and should be in `UNDERSCORE_SEPARATED_UPPERCASE`

Reply:
fixed.